src/drivers/barometer/bmp581/bmp581.h

```diff
@@ -0,0 +1,345 @@
+
+#define BMP5_SET_IIR_PRESS_MSK          (0x38)
+#define BMP5_SET_IIR_PRESS_POS          (3)
+
+#define BMP5_SHDW_SET_IIR_TEMP_MSK      (0x08)
+#define BMP5_SHDW_SET_IIR_TEMP_POS      (3)
+
+#define BMP5_SHDW_SET_IIR_PRESS_MSK     (0x20)
+#define BMP5_SHDW_SET_IIR_PRESS_POS     (5)
+
+#define BMP5_IIR_FLUSH_FORCED_EN_MSK    (0x04)
+#define BMP5_IIR_FLUSH_FORCED_EN_POS    (2)
+
+/* Powermode */
+#define BMP5_POWERMODE_MSK		(0x03)
+
+#define BMP5_DEEP_DISABLE_MSK		(0x80)
+#define BMP5_DEEP_DISABLE_POS		(7)
+
+/* Fifo configurations */
+#define BMP5_FIFO_FRAME_SEL_MSK        (0x03)
+
+/* NVM and Interrupt status asserted macros */
+#define BMP5_INT_ASSERTED_DRDY         (0x01)
+
+/*!
+ * @brief Enumerator for powermode selection
+ */
+enum bmp5_powermode {
+    /*! Standby powermode */
+    BMP5_POWERMODE_STANDBY,
+    /*! Normal powermode */
+    BMP5_POWERMODE_NORMAL,
+    /*! Forced powermode */
+    BMP5_POWERMODE_FORCED,
+    /*! Continous powermode */
+    BMP5_POWERMODE_CONTINOUS,
+    /*! Deep standby powermode */
+    BMP5_POWERMODE_DEEP_STANDBY
+};
+
+/*!
+ * @brief Enumerator for interface selection
+ */
+enum bmp5_intf {
+    /*! SPI interface */
+    BMP5_SPI_INTF,
+    /*! I2C interface */
+    BMP5_I2C_INTF,
+    /*! I3C interface */
+    BMP5_I3C_INTF
+};
+
+/*!
+ * @brief BMP5 sensor data structure which comprises of temperature and pressure in floating point with data type as
+ *  float for pressure and temperature.
+ */
+struct bmp5_sensor_data
+{
+    /*! Pressure data */
+    float pressure;
+
+    /*! Temperature data */
+    float temperature;
+};
+
+/*
+ * BMP581 internal constants and data structures.
+ */
+class IBMP581
+{
+public:
+	virtual ~IBMP581() = default;
+
+	virtual int init() = 0;
+
+	// read reg value
+	virtual uint8_t get_reg(uint8_t addr) = 0;
+
+	// bulk read reg value
+	virtual int get_reg_buf(uint8_t addr, uint8_t *buf, uint8_t len) = 0;
+
+	// write reg value
+	virtual int set_reg(uint8_t value, uint8_t addr) = 0;
+
+	virtual uint32_t get_device_id() const = 0;
+
+	virtual uint8_t get_device_address() const = 0;
+};
+
+class BMP581 : public I2CSPIDriver<BMP581>
+{
+public:
+	BMP581(const I2CSPIDriverConfig &config, IBMP581 *interface);
+	virtual ~BMP581();
+
+	static I2CSPIDriverBase *instantiate(const I2CSPIDriverConfig &config, int runtime_instance);
+	static void print_usage();
+
+	virtual int init();
+
+	void print_status();
+
+	void RunImpl();
+
+
+private:
+	static constexpr uint8_t	osr_t{BMP5_OVERSAMPLING_2X};
+	static constexpr uint8_t	osr_p{BMP5_OVERSAMPLING_32X};
+	static constexpr uint8_t	odr{BMP5_ODR_50_HZ};
+	static constexpr uint8_t	press_en{BMP5_ENABLE};
+	static constexpr uint8_t 	iir_t{BMP5_IIR_FILTER_COEFF_1};
+	static constexpr uint8_t 	iir_p{BMP5_IIR_FILTER_COEFF_1};
+
+	static constexpr uint8_t 	intr_mode{BMP5_INT_MODE_PULSED};
+	static constexpr uint8_t	intr_polarity{BMP5_INT_POL_ACTIVE_HIGH};
+	static constexpr uint8_t 	intr_drive{BMP5_INT_OD_PUSHPULL};
```

Comment:
constants should have the k prefix or full uppercase so that we know they're constants when we read the code. You should also favor longer and more descriptive names so that the reader doesn't have to hunt down the definition
```
kPressureTempOSR
```